.github/workflows/test.yml

```diff
@@ -1,65 +1,268 @@
+        with:
+          repository: nextcloud/server
+          ref: ${{ matrix.server-versions }}
+
+      - name: Checkout submodules
+        shell: bash
+        run: |
+          auth_header="$(git config --local --get http.https://github.com/.extraheader)"
+          git submodule sync --recursive
+          git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
+
+      - name: Checkout app
+        uses: actions/checkout@v2
+        with:
+          path: apps/${{ env.APP_NAME }}
+
+      - name: Set up php ${{ matrix.php-versions }}
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-versions }}
+          extensions: mbstring, iconv, fileinfo, intl, oci8
+          coverage: none
+
+      - name: Set up PHPUnit
+        working-directory: apps/${{ env.APP_NAME }}
+        run: composer i
+
+      - name: Set up Nextcloud
+        run: |
+          mkdir data
+          ./occ maintenance:install --verbose --database=${{ matrix.databases }} --database-name=XE --database-host=127.0.0.1 --database-port=1521 --database-user=autotest --database-pass=owncloud --admin-user admin --admin-pass admin
+          ./occ app:enable --force ${{ env.APP_NAME }}
+          php -S localhost:8080 &
+
+      - name: Run a smoke test
+        run: php -f ./occ files:recommendations:recommend admin
+
+  summary:
+    runs-on: ubuntu-latest
+    needs: [oci, mysql, sqlite, pgsql]
+
+    if: always()
+
+    name: test-summary
+
     steps:
-    - name: Set up php${{ matrix.php-versions }}
-      uses: shivammathur/setup-php@master
-      with:
-        php-version: ${{ matrix.php-versions }}
-        tools: composer:v1
-        extensions: ctype,curl,dom,gd,iconv,intl,json,mbstring,openssl,posix,sqlite,xml,zip
-        coverage: xdebug
-    - name: Checkout Nextcloud
-      run: git clone https://github.com/nextcloud/server.git --recursive --depth 1 -b ${{ matrix.nextcloud-versions }} nextcloud
-    - name: Install Nextcloud
-      run: php -f nextcloud/occ maintenance:install --database-host 127.0.0.1 --database-name nextcloud --database-user nextcloud --database-pass nextcloud --admin-user admin --admin-pass admin --database ${{ matrix.db }}
-    - name: Checkout app
-      uses: actions/checkout@master
-      with:
-        path: nextcloud/apps/recommendations
-    - name: Install dependencies
-      working-directory: nextcloud/apps/recommendations
-      run: composer install
-    - name: Install app
-      run: php -f nextcloud/occ app:enable recommendations
-    - name: Install app
-      run: php -f nextcloud/occ app:enable recommendations
-    - name: Run a smoke test
-      run: php -f nextcloud/occ files:recommendations:recommend admin
+      - name: Summary status
+        run: if ${{ needs.oci.result != 'success' and needs.sqlite.result != 'success' and needs.mysql.result != 'success' and needs.pgsql.result != 'success'}}; then exit 1; fi
```

Comment:
Invalid workflow file: .github/workflows/test.yml#L268
The workflow is not valid. .github/workflows/test.yml (Line: 268, Col: 14): Unexpected symbol: 'and'. Located at position 31 within expression: needs.oci